app/src/main/java/com/github/ilyashvetsov/playlistmaker/search/ui/SearchFragment.kt

```diff
@@ -68,11 +84,16 @@ class SearchActivity : AppCompatActivity() {
                 )
             }
         }
-        handleScreenState()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        binding.textSearch.removeTextChangedListener(textWatcher)
+        binding.textSearch.onFocusChangeListener = null
     }
 
     private fun handleScreenState() {
-        viewModel.screenState.observe(this@SearchActivity) { state ->
+        viewModel.screenState.observe(this@SearchFragment.viewLifecycleOwner) { state ->
```

Comment:
```suggestion
        viewModel.screenState.observe(viewLifecycleOwner) { state ->
```
Можно не указывать везде ссылку на `this`